File: src/main/java/stepanovep/fut23/utils/FutPriceUtils.java
package stepanovep.fut23.utils;

public class FutPriceUtils {

    public static int getNextBid(int startingBid, int currentBid) {
        if (currentBid != 0) {
            int inc = getIncrement(currentBid);
            return currentBid + inc;
        }

        return startingBid;
    }

    public static int getNextBid(int currentBid) {
        int inc = getIncrement(currentBid);
        return currentBid + inc;
    }

    public static int roundToValidFutPrice(int price) {
        int inc = getIncrement(price);
        return (int) ((price * 1.0 / inc) + 0.5) * inc;
    }

    private static int getIncrement(int price) {
        if (price < 1000)
            return 50;
        if (price < 10_000)
            return 100;
        if (price < 50_000)
            return 250;
        if (price < 100_000)
            return 500;
        return 1000;
    }
}

File: src/main/java/stepanovep/fut23/utils/FutPriceUtils.kt
package stepanovep.fut23.utils

class FutPriceUtils {

    companion object {
        @JvmStatic
        fun getNextBid(startingBid: Int, currentBid: Int): Int {
            if (currentBid != 0) {
                val inc = getIncrement(currentBid)
                return currentBid + inc
            }

            return startingBid
        }

        @JvmStatic
        fun getNextBid(currentBid: Int): Int {
            val inc = getIncrement(currentBid)
            return currentBid + inc
        }

        @JvmStatic
        fun roundToValidFutPrice(price: Int): Int {
            val inc = getIncrement(price)
            return ((price * 1.0 / inc) + 0.5).toInt() * inc
        }

        private fun getIncrement(price: Int): Int = when {
                price < 1000 -> 50
                price < 10_000 -> 100
                price < 50_000 -> 250
                price < 100_000 -> 500
                else -> 1000
            }
    }
}